lib/components/Atom/WheelPicker/time-wheel-picker/index.tsx

```diff
@@ -0,0 +1,321 @@
+import React, {
+    KeyboardEvent,
+    ReactNode,
+    useCallback,
+    useEffect,
+    useState,
+} from "react";
+import WheelPicker from "../base";
+import { KEY } from "@utils/common-utils";
+import { TimeWheelPickerContainerProps } from "../types";
+import dayjs from "dayjs";
+import { debounce } from "lodash";
+
+export const TimeWheelPickerContainer = ({
+    is12Hour = true,
+    startTimeIn24Format,
+    endTimeIn24Format = "23:59",
+    selectedTime = "0:00",
+    minutesInterval = 2,
+    hoursInterval = 1,
+    locale = "en-US",
+    setSelectedValue,
+    setSelectedTime,
+    close,
+    data,
+    disabled,
+}: TimeWheelPickerContainerProps) => {
+    const inputStartDate = dayjs(`1/1/1 ${startTimeIn24Format ?? "00:00"}`);
+    const inputSelectedDate = dayjs(`1/1/1 ${selectedTime}`);
+    const inputEndDate = dayjs(`1/1/1 ${endTimeIn24Format}`);
+    const [hourValue, setHourValue] = useState<{
+        label?: string | ReactNode;
+        value: string | number;
+    }>({
+        label: "00",
+        value: 0,
+    });
+    const [minuteValue, setMinuteValue] = useState<{
+        label?: string | ReactNode;
+        value: string | number;
+    }>({
+        label: "00",
+        value: 0,
+    });
+    const [meridiemValue, setMeridiemValue] = useState<{
+        label?: string | ReactNode;
+        value: string | number;
+    }>({
+        label: "AM",
+        value: "AM",
+    });
+
+    const [colRef, setColRef] = useState<boolean[]>(
+        new Array(is12Hour ? 3 : 2).fill(false),
+    );
+
+    const [minutes, setMinutes] = useState<
+        {
+            label?: string | ReactNode;
+            value: string | number;
+        }[]
+    >();
+    const [hours, setHours] = useState<
+        {
+            label?: string | ReactNode;
+            value: string | number;
+        }[]
+    >();
+    const [meridiem, setMeridiem] = useState<
+        {
+            label?: string | ReactNode;
+            value: string | number;
+        }[]
+    >();
+
+    const getHoursArray = () => {
+        const hoursData = [];
+        if (is12Hour) {
+            for (let i = 1; i <= 12; i += hoursInterval) {
+                hoursData.push({
+                    label: `0${i}`.slice(-2),
+                    value: i,
+                });
+            }
+        } else {
+            for (
+                let i = inputStartDate.get("hour");
+                i <= (inputEndDate.get("hour") || 23);
+                i += hoursInterval
+            ) {
+                hoursData.push({
```

Comment:
Seems that in cycle we do the same thing 
```
                hoursData.push({
                    label: `0${i}`.slice(-2),
                    value: i,
                });
```
And the only difference between cycles is initial value and conditions for breaking cycle, which depends on `is12Hour`. Can we make it shorter a bit?